File: foundations-of-programming/src/main/java/Interpreter/Interpreter.java
package Interpreter;

public class Interpreter {
    public int interpret(int value, String[] commands, int[] args) {
        int result = value;
        for (int i = 0; i < commands.length; i++) {
            if (!commands[i].matches("[-+*]")) {
                return -1;
            }
            result = operation(commands[i], result, args[i]);
        }
        return result;
    }

    private int operation(String command, int value1, int value2) {
        switch (command) {
            case "*":
                return value1 * value2;
            case "+":
                return value1 + value2;
            case "-":
                return value1 - value2;
            default:
                return -1;
        }
    }
}

File: foundations-of-programming/src/main/java/Interpreter/interpreter.py
import re


def interpret(value, commands, args):
    result = value
    for i in range(len(commands)):
        match = re.match('[-+*]', commands[i], re.M | re.I)
        if match:
            result = operation(commands[i], result, args[i])
        else:
            return -1
    return result


def operation(command, value1, value2):
    if command == '+':
        return value1 + value2
    elif command == '-':
        return value1 - value2
    elif command == '*':
        return value1 * value2
    else:
        return -1
